bool a
bool b
bool x
bool y